### PR TITLE
mgr/dashboard: Improve the size of forms

### PR DESCRIPTION
**Pull request by tspmelo**
**mgr/dashboard: Use custom classes for form label and inputs**
With this change we are able to configura the col sizes of labels and inputs in
one single place.

Also increased the col size in smaller screens to be able to display longer
labels.

This was not applied in some situations where the col size was different than
usual.

**mgr/dashboard: Add required symbol directly in the text element**
This prevents the symbol from breaking from the last word of the label,
if the width gets too small.

This will also remove the need to add an extra span just to show the symbol.

Fixes: https://tracker.ceph.com/issues/40751

Before:

After:

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check arm64

**Comment by Devp00l** [comment]
Hi, I've tested the changes locally but the mobile view looks better to me in the current master.

I made some screenshots to look at the differences.

For a size of 568px:

Yours
form-mobile-view-tiago

Current master
form-mobile-view-master


For a size of 377px:

Yours
form-mobile-view-377px-tiago

Current master
form-mobile-view-377px-master

**Comment by LenzGr** [comment]
With regards to the minimum screen size, we have discussed this before and noted in [issues#41941](https://tracker.ceph.com/issues/41941).

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test docs

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html [hunk 1]
```diff
@@ -11,12 +11,10 @@
       <div class="card-body">
         <!-- Target IQN -->
         <div class="form-group row">
-          <label class="col-form-label col-sm-3"
-                 for="target_iqn">
-            <ng-container i18n>Target IQN</ng-container>
-            <span class="required"></span>
-          </label>
-          <div class="col-sm-9">
+          <label class="cd-col-form-label required"
```
Comment (s0nea) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html]: Does this also need to be applied to e.g. https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-discovery-modal/iscsi-target-discovery-modal.component.html#L13 ?

Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html]: Done.

**Comment by tspmelo** [comment]
I added a new scss rule so the labels in modals don't get to small:
```
cd-modal {
  .cd-col-form-label {
    @extend .col-lg-4;
  }

  .cd-col-form-input {
    @extend .col-lg-8;
  }
}
```

**Pull request merged**